edb/schema/objects.py

```diff
@@ -1764,8 +1811,11 @@ def _reduce_to_ref(  # type: ignore
 class ObjectIndexBase(ObjectCollection, container=tuple):
     _key: KeyFunction
 
-    def __init_subclass__(cls, *, key: KeyFunction):
-        cls._key = key
+    def __init_subclass__(cls, *, key: Optional[KeyFunction] = None) -> None:
+        if key is not None:
+            cls._key = key
+        elif cls._key is None:
+            raise TypeError('missing required "key" class argument')
```

Comment:
Not clear to me why this is necessary.

Reply:
ObjectIndex must define a `key` function, but once it's defined, subclasses don't need to redefine it.

Reply:
This includes parametrized subclasses, obviously.

Reply:
Oh, of course, I didn't read it properly myself. Thanks!